src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -990,18 +1214,416 @@ partial class C
 
                 partial class C
                 {
-                    public partial int P { get => throw null!; set { } }
+                    public partial int P { internal get => throw null!; set { } }
                 }
                 """;
 
             var comp = CreateCompilation(source);
 
-            // PROTOTYPE(partial-properties): missing diagnostic
-            comp.VerifyEmitDiagnostics();
+            comp.VerifyEmitDiagnostics(
+                // (8,37): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+                //     public partial int P { internal get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "get").WithLocation(8, 37),
+                // (8,57): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+                //     public partial int P { internal get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "set").WithLocation(8, 57));
+        }
+
+        [Fact]
+        public void ModifierDifference_Readonly_Property()
+        {
+            // readonly modifier mismatch on property
+            var source = """
+                partial struct S
+                {
+                    readonly partial int P1 { get; set; }
+                    partial int P2 { get; set; }
+                    readonly partial int P3 { get; set; }
+                }
+
+                partial struct S
+                {
+                    partial int P1 { get => throw null!; set { } }
+                    readonly partial int P2 { get => throw null!; set { } }
+                    readonly partial int P3 { get => throw null!; set { } }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+
+            comp.VerifyEmitDiagnostics(
+                // (10,17): error CS8663: Both partial member declarations must be readonly or neither may be readonly
+                //     partial int P1 { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberReadOnlyDifference, "P1").WithLocation(10, 17),
+                // (11,26): error CS8663: Both partial member declarations must be readonly or neither may be readonly
+                //     readonly partial int P2 { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberReadOnlyDifference, "P2").WithLocation(11, 26));
+        }
+
+        [Fact]
+        public void ModifierDifference_Readonly_Accessors()
+        {
+            // readonly modifier mismatch on accessors
```

Comment:
Consider testing the third case, where the `readonly` modifiers match

Reply:
I grouped the tests by modifier in order to make it easier to check if we have covered both happy and sad cases.